src/plasmapy/simulation/particle_tracker/particle_tracker.py

```diff
@@ -706,6 +799,55 @@ def _push(self) -> None:
 
         self.dt = dt
 
+        # Update velocities to reflect stopping
+        if not self._do_stopping:
+            return
+
+        # TODO: maybe break out different stopping models into separate functions?
+        #  particle_stopping_routines?
+        # S = np.zeros(self.nparticles_tracked) * u.J / u.m
+        #
+        # for grid in self.grids:
+        #     _S = grid.nearest_neighbor_interpolator(
+        #         pos_tracked * u.m, "S", persistent=False
+        #     )
+        #     _S = np.nan_to_num(_S, nan=0.0 * u.J / u.m)
+        #
+        #     S += _S
+        speeds = np.linalg.norm(self.v[tracked_mask], axis=-1)[:, np.newaxis]
+        dx = np.multiply(speeds, dt)
+        stopping_power = np.zeros((self.nparticles_tracked, 1))
+
+        for cs in self._stopping_power_interpolators:
+            # TODO: is this the proper way to handle the addition of multiple stopping powers?
```

Comment:
This should be about right! Stopping is really particle-particle interactions: the lattice structure etc. of the material only matters insofar as it affects the density and thus the probability of a collision. So, if you have a mixture of gold and aluminum, say, I think the decrease in energy should just add, e.g. 

dE = (sp1*rho1 + sp2*rho2)*dx